fursquare/venues/views.py

```diff
@@ -0,0 +1,332 @@
+
+    elif request.method == 'DELETE':
+        comment.delete()
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+@api_view(['GET', 'POST'])
+@permission_classes((IsAdminOrReadOnly, ))
+def user_list(request):
+    if request.method == 'GET':
+        user_list = User.objects.all()
+        serializer = UserSerializer(user_list, many=True)
+        return Response(serializer.data)
+
+    elif request.method == 'POST':
+        data = request.data
+        serializer = UserSerializer(data=data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+@api_view(['GET', 'PATCH', 'PUT', 'DELETE'])
+@permission_classes((IsAdminOrReadOnly, ))
+def user_detail(request, pk):
+    try:
+        user = User.objects.get(pk=pk)
+    except User.DoesNotExist:
+        return Response(status=status.HTTP_404_NOT_FOUND)
+
+    if request.method == 'GET':
+        serializer = UserSerializer(user)
+        return Response(serializer.data)
+
+    elif request.method == 'PATCH':
+        data = request.data
+        serializer = UserSerializer(user, data=data, many=True, partial=True)
+        if serializer.valid():
+            serializer.save()
+            return Response(serializer.data)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    elif request.method == 'PUT':
+        data = request.data
+        serializer = UserSerializer(user, data=data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    elif request.method == 'DELETE':
+        user.delete()
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+
+@api_view(['GET', 'POST'])
+@permission_classes((IsAdminOrReadOnly, ))
+def rating_list(request):
+    if request.method == 'GET':
+        rating_list = Rating.objects.all()
+        serializer = RatingSerializer(rating_list, many=True)
+        return Response(serializer.data)
+
+    elif request.method == 'POST':
+        user = request.user
+        data = request.data
+        data['rated_by'] = user.id
+
+        serializer = RatingSerializer(data=data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+@api_view(['GET', 'POST'])
+@permission_classes((IsAdminOrReadOnly, ))
+def venue_rating_list(request, pk):
+
+    try:
+        venue = Venue.objects.get(pk=pk)
+    except Venue.DoesNotExist:
+        return Response(status=status.HTTP_404_NOT_FOUND)
+    ratings = venue.get_venue_ratings
+    if request.method == 'GET':
+        print(ratings)

```

Comment:
GENEL UYARI  - LUTFEN PRINT IFADELERI SILINSIN.

Reply:
PRINT ILE DEBUG YAPILMAZ. IKI OPSIYONUN VAR. YA PDB-IPDB KURACAKSIN YA DA PYCHARM'IN DEBUGGER'INI KULLANACAKSIN.